Tests/Services/HealthTests/HealthTests.swift

```diff
@@ -0,0 +1,236 @@
+      }
+
+      group.addTask {
+        try await client.run()
+      }
+
+      do {
+        try await body(healthClient, health.provider)
+      } catch {
+        XCTFail("Unexpected error: \(error)")
+      }
+
+      group.cancelAll()
+    }
+  }
+
+  func testCheckOnKnownService() async throws {
+    try await withHealthClient { (healthClient, healthProvider) in
+      let testServiceDescriptor = ServiceDescriptor(package: "test.package", service: "TestService")
+
+      try healthProvider.updateService(
+        descriptor: testServiceDescriptor,
+        status: .serving
+      )
+
+      try healthProvider.updateService(
+        descriptor: ServiceDescriptor(package: "package.to.be.ignored", service: "IgnoredService"),
+        status: .notServing
+      )
+
+      var message = Grpc_Health_V1_HealthCheckRequest()
+      message.service = testServiceDescriptor.fullyQualifiedService
+
+      try await healthClient.check(request: ClientRequest.Single(message: message)) { response in
+        try XCTAssertEqual(response.message.status, .serving)
+      }
+    }
+  }
+
+  func testCheckOnUnknownService() async throws {
+    try await withHealthClient { (healthClient, healthProvider) in
+      try healthProvider.updateService(
+        descriptor: ServiceDescriptor(package: "package.to.be.ignored", service: "IgnoredService"),
+        status: .notServing
+      )
+
+      var message = Grpc_Health_V1_HealthCheckRequest()
+      message.service =
+        ServiceDescriptor(package: "does.not", service: "Exist").fullyQualifiedService
+
+      try await healthClient.check(request: ClientRequest.Single(message: message)) { response in
+        try XCTAssertThrowsError(response.message) { error in
+          XCTAssertTrue(error is RPCError)
+          XCTAssertEqual((error as! RPCError).code, .notFound)
+        }
+      }
+    }
+  }
+
+  func testWatchOnKnownService() async throws {
+    try await withHealthClient { (healthClient, healthProvider) in
+      let testServiceDescriptor = ServiceDescriptor(package: "test.package", service: "TestService")
+      let ignoredServiceDescriptor = ServiceDescriptor(
+        package: "package.to.be.ignored",
+        service: "IgnoredService"
+      )
+
+      let statuses: [ServingStatus] = [.serving, .notServing, .serving, .serving, .notServing]
+
+      try healthProvider.updateService(
+        descriptor: testServiceDescriptor,
+        status: statuses[0]
+      )
+
+      try healthProvider.updateService(
+        descriptor: ignoredServiceDescriptor,
+        status: .notServing
+      )
+
+      var message = Grpc_Health_V1_HealthCheckRequest()
+      message.service = testServiceDescriptor.fullyQualifiedService
+
+      try await healthClient.watch(request: ClientRequest.Single(message: message)) { response in
+        var responseStreamIterator = response.messages.makeAsyncIterator()
+
+        for i in 0 ..< statuses.count {
+          let next = try await responseStreamIterator.next()!
```

Comment:
Avoid '!'-ing here. There's plenty of ways that this _could_ go wrong and we'd want to avoid a crash if it did.

You can use `let message = try XCTUnwrap(next)` to unwrap the optional or throw an error (and fail the test) if it's nil.